File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>mpp;
        int n=nums.size();
        for(int i=0;i<n;i++)
        {
            int num=nums[i];
            int rem=target-num;
            if(mpp.find(rem) !=mpp.end())
            return {mpp[rem],i};
            mpp[num]=i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hmap = {}
        for i,n in enumerate(nums):
            val = target - n
            if val in hmap:
                return [hmap[val] , i]
            hmap[n] = i
        return

            
            
        